tuxemon/log.py

```diff
@@ -72,8 +98,78 @@ def configure():
             log_hdlr.setFormatter(logging.Formatter("%(asctime)s - %(name)s - " "%(levelname)s - %(message)s"))
             logger.addHandler(log_hdlr)
 
+            # Archive previous log
+            archive_log()
+            # Logging to file
+            log_filehandler = logging.FileHandler(f"{USER_LOG_DIR}/latest.log")
+            log_filehandler.setLevel(logging.INFO)
+            log_filehandler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - " "%(levelname)s - %(message)s"))
+            logger.addHandler(log_filehandler)
+
             loggers[logger_name] = logger
 
             # prevent pyscroll redraw warnings
             pyscroll_logger = logging.getLogger("orthographic")
             pyscroll_logger.setLevel(logging.ERROR)
+
+def send_logs():
+    config = prepare.CONFIG
+    logging.info("="* 26)
+
+    logging.info("=== System information ===")
+
+    # OS Info
+    entries = {
+        platform.platform: "platform",
+        platform.processor: "processor",
+        platform.python_version: "python_version",
+        platform.python_implementation: "python_implementation",
+    }
+    for i in entries:
+        logging.info(f"{entries[i]}\t{i()}")
+
+    logging.info("=== Library versions ===")
+
+    lib_entries = {
+        "babel": "babel",
+        "cbor": "cbor",
+        "neteria": "neteria",
+        "pillow": "PIL",
+        "pygame": "pygame",
+        "pyscroll": "pyscroll",
+        "pytmx": "pytmx",
+        "requests": "requests",
+        "natsort": "natsort",
+        "PyYaml": "yaml"
+    }
+    for i in lib_entries:
+        try:
+            logging.info(f"{i}\t{__import__(lib_entries[i]).__version__}")
+        except:
+            logging.info(f"{i}\tunknown")
+
+    try:
+        logging.info(f"pygame_sdl_version\t{__import__('pygame').get_sdl_version()}")
+    finally:
+        pass
+
+    # Get file
+    file = open(f"{USER_LOG_DIR}/latest.log")
+    send_files = {"tuxemon_log.txt": file}
+    r = requests.post(config.log_host_url, files=send_files, headers={"Max-Days": config.log_storage_max_days, "User-Agent": "curl/7.80.0"})
```

Comment:
im guessing that the header here is specific to transfer.sh?  to prevent leaking logs without proper expiration, lets either make a generic class that handles sending logs to a provider, or simply raise an error if its not going to transfer.sh, and put a TODO here about allowing other providers